### PR TITLE
Xp branch merge

### PR DESCRIPTION
**Pull request by JasonIqbal**
Can you review this branch for merging? The changes include added color, menu, highscore, history, game state, and a start game flash message.

### CODE REVIEW

**Comment by sterlinghirsh** [comment]
Wow I haven't had a chance to look over this completely yet but this looks awesome! Glad you're still thinking about this too :)

**Comment by addison-grant** [comment]
I'll look it over and test it by Sunday.

**Comment by JasonIqbal** [comment]
thank you

File: src/App.js [hunk 1]
```diff
@@ -4,16 +4,43 @@ import House from './components/House.js';
...
+  constructor(props) {
+        super(props);
+        this.state = {gamestate: 'game'};
+        
   }
+  
+  // ether bind or use es6
+  changeGameState = (newstate) => {
+      console.assert(newstate === 'game' || newstate === 'menu',
+       "Cannot set game state invalid arg input.");
```
Comment (sterlinghirsh) [inline src/App.js]: I would usually put the bad input in the error message to make it easier to debug

File: src/components/House.js [hunk 2]
```diff
@@ -18,30 +18,33 @@ import Rug from './Rug.js';
...
 
 class House extends React.Component {
+    
   constructor(props) {
     super(props);
 
     this.state = {
       rows: 6,
       columns: 8,
+      gameIntervalID: '',
```
Comment (sterlinghirsh) [inline src/components/House.js]: Might want to init this as null instead

File: src/components/Score.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+class Score {
+    constructor() {
+        this.score = 0;
+    }
+
+    add(increment) {
+        console.assert(increment >= 0, this.Score + increment !== Infinity);
```
Comment (sterlinghirsh) [inline src/components/Score.js]: Why is it wrong to reduce the score?

File: src/components/House.js [hunk 4]
```diff
@@ -31,7 +31,8 @@ class House extends React.Component {
       maxScores: {},
       startDate: new Date(),
       currentDate: new Date(),
-      repairCount: 0
+      repairCount: 0,
+      age: ''
```
Comment (sterlinghirsh) [inline src/components/House.js]: Hm, is this used?

File: src/components/Score.js (hunk 3 above)
Comment (JasonIqbal) [reply src/components/Score.js]: it isn't wrong, it is just not needed. If a person has repaired something how can that type of count be reversed? If we change the score metrics to something else then definitely and we should write a decrement function or modify this to do both.

File: src/App.js (hunk 1 above)
Comment (JasonIqbal) [reply src/App.js]: This is true, it does help. Unless the error is the end of string null terminator missing.

**Comment by JasonIqbal** [comment]
With gamestate we can add new levels and cool intermission videos or animations like pac-man does or whatever we like, such as extra menu item screens.

File: src/App.js (hunk 1 above)
Comment (JasonIqbal) [reply src/App.js]: I also want to first check the security risk, if any. I want to be very careful with variables when I don't know what they contain. Have you heard of that sql hack that programmer did a long time ago where the user's input was restructured in an sql command for searching. He was able to insert his own request and have the whole table output.

Comment (sterlinghirsh) [reply src/App.js]: I'm familiar with sql injection attacks, and good on you for watching out for injection vulnerabilities, but I think in this case we're not so worried. We don't usually hit the null termination issue in js, that's more of a c thing.

**Comment by sterlinghirsh** [comment]
Hey this is great! I made some changes and merged it.

**Comment by JasonIqbal** [comment]
Thank you

**Pull request merged**